bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -177,6 +177,7 @@ void RequestsBatch::releaseReqsAndSendBatchedReplyIfCompleted(PreProcessReplyMsg
   string batchCid;
   atomic_uint32_t batchSize = 0;
   PreProcessBatchReplyMsgSharedPtr batchReplyMsg;
+  set<uint16_t> reqOffsetsInBatch;
```

Comment:
Nit: Can we use another container which has better insertion time complexity ,here e.g. a vector? which gives amortized constant time insertion complexity while set will give logarithmic to the size of set time complexity?

Reply:
I can change that, but for a few tens of requests, it really does not matter.